tests/functional/metadata/mongodb/listObject.spec.js

```diff
@@ -523,6 +536,23 @@ describe('MongoClientInterface::metadata.listObject', () => {
                     return done();
                 });
             });
+
+            it('Should not include location in listing result and use custom listing parser', done => {
+                const defaultListingParser = metadata._listingParser;
```

Comment:
Instead of modifying the parser of the object (and having to reset it), you should create a new client, to keep the tests independents (and possibly allow running them in parallel):

```suggestion
                const metadata = new MetadataWrapper(IMPL_NAME, opts, null, logger);
                metadata.setup(done);
```